nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/data/explorer/PutAzureDataExplorer.java

```diff
@@ -0,0 +1,309 @@
+                    .append("pollOnIngestionStatus - ").append(pollOnIngestionStatus).append("|")
+                    .append("ingestionStatusPollingTimeout - ").append(ingestionStatusPollingTimeout).append("|")
+                    .append("ingestionStatusPollingInterval - ").append(ingestionStatusPollingInterval).append("|")
+                    .append("routePartiallySuccessfulIngestion - ").append(routePartiallySuccessfulIngestion);
+            getLogger().debug(ingestLogString.toString());
+
+            KustoIngestionResult result = service.ingestData(
+                    new KustoIngestionRequest(streamingEnabled, pollOnIngestionStatus, inputStream, databaseName,
+                            tableName, dataFormat, mappingName, ignoreFirstRecord, ingestionStatusPollingTimeout, ingestionStatusPollingInterval)
+            );
+            getLogger().info("Ingest Status Polling Enabled {} and Ingest Status Polling Enabled {} ", pollOnIngestionStatus, result.toString());
+
+            if (result == KustoIngestionResult.SUCCEEDED) {
+                getLogger().info("Ingest Completed - {}", result.getStatus());
+            } else if (result == KustoIngestionResult.FAILED) {
+                getLogger().error("Ingest Failed - {}", result.getStatus());
+                isError = true;
+            } else if (result == KustoIngestionResult.PARTIALLY_SUCCEEDED) {
+                getLogger().warn("Ingest Partially succeeded - {}", result.getStatus());
+                flowFile = session.putAttribute(flowFile, "ingestion_status", "partial_success");
+                if (StringUtils.equalsIgnoreCase(routePartiallySuccessfulIngestion, "Failure")) {
+                    isError = true;
+                }
+            }
+
+        } catch (IOException | URISyntaxException e) {
+            getLogger().error("Azure Data Explorer Ingest processing failed", e);
+            isError = true;
+        }
+
+        if (isError) {
+            getLogger().error("Ingest processing failed {}", flowFile);
+            session.transfer(flowFile, FAILURE);
+        }else {
+            getLogger().info("Ingest processing completed {}", flowFile);
+            session.transfer(flowFile, SUCCESS);
+        }
+    }
+
+    protected boolean checkIfStreamingPolicyIsEnabledInADX(String entityName, String database) {
+        KustoQueryResponse kustoQueryResponse = service.executeQuery(database, String.format(STREAMING_POLICY_SHOW_COMMAND, PutAzureDataExplorer.DATABASE, entityName));
+        if (kustoQueryResponse.isError()) {
+            throw new ProcessException(String.format("Error occurred while checking if streaming policy is enabled for the table for entity %s in database %s",entityName,database));
+        }
+        List<List<Object>> queryResult = kustoQueryResponse.getQueryResult();;
+        if ( queryResult.get(0) !=null && queryResult.get(0).get(2)!=null && StringUtils.isNotEmpty(queryResult.get(0).get(2).toString())) {
```

Comment:
This conditional is difficult to follow based on the construction and formatting. The `List<List<Object>>` structure is also very generic, which makes this difficult to understand. If this is a value that should always be checked, recommend evaluating the value in the service and providing a separate named property.